core/src/autogluon/core/utils/utils.py

```diff
@@ -484,23 +557,20 @@ def generate_train_test_split(
         if stratify is None:
             raise
         X_train, X_test, y_train, y_test = train_test_split(
-            X_split, y_split.values, test_size=test_size, shuffle=True, random_state=random_state, stratify=None
+            X_split, y_split.values, test_size=test_size, train_size=train_size, shuffle=True, random_state=random_state, stratify=None
         )
         if len(y_test) >= len(y_split.unique()):
             # This should never occur, otherwise the original exception is not an expected one
             raise
-    if problem_type != SOFTCLASS:
-        y_train = pd.Series(y_train, index=X_train.index)
-        y_test = pd.Series(y_test, index=X_test.index)
-    else:
-        y_train = pd.DataFrame(y_train, index=X_train.index)
-        y_test = pd.DataFrame(y_test, index=X_test.index)
+    cls_y = type(y)
```

Comment:
qq: why do we remove the check here?

Reply:
Because this check is really about the type of y_train/y_test. SOFTCLASS is the only problem type that is in the DataFrame format. The new logic generalizes and removes unnecessary hardcoding.